subsys/shell/backends/Kconfig.backends

```diff
@@ -187,6 +187,102 @@ source "subsys/logging/Kconfig.template.log_config"
 
 endif # SHELL_BACKEND_RTT
 
+config SHELL_BACKEND_MQTT
+	bool "Enable MQTT backend."
+	depends on NET_TCP
+	depends on NET_IPV4
+	depends on NETWORKING
+	select DNS_RESOLVER
+	select HWINFO
+	select MQTT_LIB
+	select NET_MGMT
+	select NET_MGMT_EVENT
+	help
+	  Enable MQTT backend.
+
+if SHELL_BACKEND_MQTT
+
+config SHELL_MQTT_SERVER_ADDR
+	string "MQTT server address"
+	default "192.168.0.100"
+	help
+	  MQTT server address.
+
+config SHELL_MQTT_SERVER_PORT
+	int "MQTT server port"
+	default 1883
+	help
+	  MQTT server port.
+
+config SHELL_MQTT_SERVER_USERNAME
+	string "MQTT server username"
+	help
+	  MQTT server username.
+
+config SHELL_MQTT_SERVER_PASSWORD
+	string "MQTT server password"
+	help
+	  MQTT server password.
+
+config SHELL_MQTT_RX_BUF_SIZE
+	int "RX buffer size"
```

Comment:
Do we need a range for RX too?

Reply:
The RX side uses `uint32_t` or `sizt_t`, I think we don't need range for this?

Reply:
I was just wondering how the system will work if user will put a really large value here.